common-custom-user-data-maven-extension/src/main/java/com/gradle/maven/common/user/data/extension/CommonCustomUserDataMavenExtensionEventSpy.java

```diff
@@ -0,0 +1,37 @@
+package com.gradle.maven.common.user.data.extension;
+
+import org.apache.maven.eventspy.AbstractEventSpy;
+import org.apache.maven.eventspy.EventSpy;
+import org.codehaus.plexus.PlexusContainer;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.logging.Logger;
+
+import javax.inject.Inject;
+
+@Component(
```

Comment:
Is this event spy required ? Can't we call `ApiAccessor.registerExtension(getClass(), container, logger);` inside the constructor of the `CommonCustomUserDataGradleEnterpriseListener` where we do have access to the `PlexusContainer` as well?

Reply:
There are cases where the `CommonCustomUserDataGradleEnterpriseListener` is not constructed, even if marked with the `@Component` annotation.
But even there, it needs the `GradleEnterpriseListener` interface at runtime when constructed, so the interface needs to be available in its realm when constructed.(and luckily, it's constructed after `EventSpy#init` is called)